pimd/pim_nht.c

```diff
@@ -576,6 +612,15 @@ static int pim_ecmp_nexthop_search(struct pim_instance *pim,
 			pim_addr nhaddr = nh_node->gate.ipv6;
 #endif
 			nbrs[i] = pim_neighbor_find(ifps[i], nhaddr);
+#if PIM_IPV == 6
```

Comment:
Since whenever pim_neighbor_find api fails you are trying to call pim_neighbor_find_by_secondary function everywhere, I think it will be better to move this section of code inside pim_neighbor_find api so you don't need to change code wherever the api is called. This will avoid duplication of code as well.

Reply:
Addressed.